### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by Malivil**
**Elementalist**
- Fixed elementalist's upgraded midnight overlay not activating if it triggered on the first shot
- Fixed elementalist's upgraded midnight overlay not being full-black
- Fixed elementalist's midnight ability always going full black instead of dimming based on damage

**Soulbound**
- Fixed error in soulbound ability when beenade not installed

### CODE REVIEW

File: lua/customroles/elementalist/cl_elementalist.lua [hunk 1]
```diff
@@ -2,8 +2,15 @@
 local shouldDrawdarkOverlay = false
 local iceOverlay = nil
 local darkOverlay = {
-    ["$pp_colour_brightness"] = 1,
-    ["$pp_colour_colour"] = 1
+    [ "$pp_colour_addr" ] = 0,
```
Comment (Malivil) [inline lua/customroles/elementalist/cl_elementalist.lua:5]: `DrawColorModify` requires full table even though the structure it takes claims there are default values.
This fixes the blind always being full black.

File: lua/customroles/elementalist/cl_elementalist.lua [hunk 2]
```diff
@@ -45,29 +52,25 @@ local function CloseIceOverlay()
 end
 
 local function StartBlindOverlay()
-    darkOverlay = {
-        ["$pp_colour_brightness"] = -1,
-        ["$pp_colour_colour"] = 0
-    }
+    shouldDrawdarkOverlay = true
```
Comment (Malivil) [inline lua/customroles/elementalist/cl_elementalist.lua:55]: This fixes an edge case where the upgraded midnight wouldn't trigger the blind on the first shot, even if it procced

File: lua/customroles/elementalist/cl_elementalist.lua [hunk 3]
```diff
@@ -86,7 +89,7 @@ net.Receive("EndIceScreen", function(len)
 end)
 
 net.Receive("BeginDimScreen", function(len)
-    local amount = net.ReadUInt(6)
+    local amount = net.ReadUInt(7)
```
Comment (Malivil) [inline lua/customroles/elementalist/cl_elementalist.lua:92]: This allows values over 64, fixing the full black from the upgraded midnight

**Review by NoxxFlame — APPROVED** [review]
LGTM

**Pull request merged**